### PR TITLE
(Menu) Option to display clock at all times, even in-game

### PR DESCRIPTION
**Pull request by grimpunch**
## Related Issues
https://github.com/libretro/RetroArch/issues/6890

### CODE REVIEW

File: gfx/video_driver.c [hunk 1]
```diff
@@ -2464,7 +2464,7 @@ void video_driver_frame(const void *data, unsigned width,
          if (video_info.fps_show)
          {
             snprintf(video_info.fps_text, sizeof(video_info.fps_text),
-                  "||  FPS: %6.1f ", last_fps);
+                  "|| FPS: %6.1f ", last_fps);
```
Comment (fr500) [inline gfx/video_driver.c]: That space was there on purpose so it looked uniform

File: msg_hash.h [hunk 2]
```diff
@@ -1111,106 +1111,107 @@ enum msg_hash_enums
...
-
-   MENU_LABEL(NETPLAY_GAME_WATCH),
-   MENU_LABEL(CHEAT_INDEX_MINUS),
-   MENU_LABEL(CHEAT_INDEX_PLUS),
-   MENU_LABEL(SHADER_NEXT),
-   MENU_LABEL(SHADER_PREV),
-   MENU_LABEL(FRAME_ADVANCE),
-   MENU_LABEL(FPS_SHOW),
-   MENU_LABEL(STATISTICS_SHOW),
+MENU_ENUM_LABEL_VALUE_NOT_AVAILABLE,
```
Comment (RobLoach) [inline msg_hash.h]: Undesired whitespace change? Looks strange against the rest of the code.

**Comment by RobLoach** [comment]
Interesting idea... Could this use the On-Screen Display somehow? Unsure how the components UI works, but this seems like a good idea for people who want it.

File: gfx/video_driver.c (hunk 1 above)
Comment (twinaphex) [reply gfx/video_driver.c]: Yeah, please change that part back.

**Comment by twinaphex** [comment]
Seems like a fine option. Only thing I want corrected first is the small bit about the extra space that was added before the 'FPS' text. After that I feel this is fine to merge.

As for a more fundamental rewriting of On-Screen Display, I think that falls beyond the scope of this PR. Let's first focus on getting this part merged.

**Comment by grimpunch** [comment]
Fixed the whitespace issue with FPS text. @twinaphex 
Please let me know if there are any more issues. :+1:

**Comment by Ryunam** [comment]
Thank you for implementing this! I will make sure to test it shortly and report back here.

**Comment by orbea** [comment]
@grimpunch I rebased this PR and added some missing bits, but I noticed one remaining problem I am not sure how to solve. The time conflicts with the fps and framecount and will not display if either of those are displayed.

There are two ways I can guess may fix this, either it should be better integrated with the fps and framecount to display right of those if they are enabled or it should be made more like the the OSD statitistics and display on another part of the screen, perhaps the bottom right corner?

The rebased branch can be found here, is there any chance you can help resolve this last issue?

https://github.com/orbea/RetroArch/tree/time

**Comment by grimpunch** [comment]
I agree with Alfredo, I'll pick this up and rebase, then refactor the
functionality into a widget when they are ready.

**Comment by RobLoach** [comment]
Looks like there's some conflicts.

**Comment by ofry** [comment]
@grimpunch Please rebase this PR :+1:

**Comment by grimpunch** [comment]
Ready again

**Comment by grimpunch** [comment]
@ofry

**Comment by RobLoach** [comment]
aaaa

Two questions...

1. There are two Display Frame Count entries?
2. When Display FPS is enabled, the time no longer shows. May want to selectively append to that entry?

**Comment by ofry** [comment]
@grimpunch Please rebase this PR to solve conflicts.

**Comment by twinaphex** [comment]
@jdgleaver Any ideas on how to move forward here? Do we reimplement it from scratch, do we do it through widgets, etc?

**Comment by jdgleaver** [comment]
The last I heard, @Ryunam was still working on this (in a follow-up to https://github.com/libretro/RetroArch/pull/10526) - but I'm more than happy to take it over. I'm planning to overhaul the widget-based status display (FPS, etc.) anyway, and adding a clock would be a natural extension. I have a pretty full schedule next week, but can look into implementing this soon after.

**Comment by grimpunch** [comment]
Hi! I saw @Ryunam took this and refactored it for further work, going to close this one now

**Pull request closed without merging**